staging/src/k8s.io/client-go/util/flowcontrol/backoff.go

```diff
@@ -33,6 +33,7 @@ type backoffEntry struct {
 type Backoff struct {
 	sync.RWMutex
 	Clock           clock.Clock
+	HasExpiredFunc  func(eventTime time.Time, lastUpdate time.Time, maxDuration time.Duration) bool
```

Comment:
needs docs. Any semantics expected from this func?

Reply:
Great place to document the pre-existing behavior with a "if nil, ..."

Reply:
> Any semantics expected from this func?

I'm not sure I know what you mean by this question, but the clients of this util are many and highly unrelated (even within kubelet). Overall I tried to inject this as independently as possible and with as little NewBackoff signature change as possible.

Reply:
Fixed docs in 6d33dda63b656e63e9eef41194b9bbd9f65a2b13